src/workload.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python3
+
+# Copyright 2024 Canonical Limited
+# See LICENSE file for licensing details.
+
+"""This model contains classes and methods related to Kyuubi workload."""
+
+import socket
+
+from ops.model import Container
+
+from constants import JDBC_PORT, KYUUBI_CONTAINER_NAME, KYUUBI_SERVICE_NAME, SPARK_PROPERTIES_FILE
+from models import User
+from utils import ContainerFile, IOMode, WithLogging
+
+
+class KyuubiServer(WithLogging):
+    """The abstraction of Kyuubi workload container."""
+
+    def __init__(self, container: Container, user: User = User()):
+        self.container = container
+        self.user = User
+
+    def get_spark_configuration_file(self, mode: IOMode) -> ContainerFile:
+        """Return the configuration file for Spark History server."""
+        daemon_user = User(name="_daemon_", group="_daemon_")
```

Comment:
**todo** why not using `self.user`? I believe you are now using the `self.user`, you could just replace the fact that we use "spark" with using `_daemon_`?

**out-of-scope, thought** Maybe we could consider to run the pebble services (not pebble) as the spark user. But let's think about this and discuss it first

Reply:
Updated to use `self.user` while the `User` class has `_daemon_` as default values for `user` and `group`.